src/views/W3NCreateForm/W3NCreateForm.tsx

```diff
@@ -0,0 +1,110 @@
+import { useCallback, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { browser } from 'webextension-polyfill-ts';
+
+import { Web3Names } from '@kiltprotocol/did';
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers';
+import { u32 } from '@polkadot/types';
+
+import * as styles from './W3NCreateForm.module.css';
+
+import { Identity } from '../../utilities/identities/types';
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import { CopyValue } from '../../components/CopyValue/CopyValue';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { Stats } from '../../components/Stats/Stats';
+
+interface Props {
+  identity: Identity;
+  onSubmit: (name: string) => void;
+}
+
+export function W3NCreateForm({ identity, onSubmit }: Props): JSX.Element {
+  const t = browser.i18n.getMessage;
+  const { goBack } = useHistory();
+
+  const [error, setError] = useState('');
+  const handleInput = useCallback(() => setError(''), []);
+
+  const handleSubmit = useCallback(
+    async (event) => {
+      event.preventDefault();
+
+      const formData = new FormData(event.target as HTMLFormElement);
+      const value = formData.get('name') as string;
+      const name = value.trim();
+
+      const { api } = await BlockchainApiConnection.getConnectionOrConnect();
+      const minLength = (api.consts.web3Names.minNameLength as u32).toNumber();
+      const maxLength = (api.consts.web3Names.maxNameLength as u32).toNumber();
+
+      const tooShort = name.length < minLength;
+      if (tooShort) {
+        setError(t('view_W3NCreateForm_short', minLength));
+        return;
+      }
+
+      const tooLong = name.length > maxLength;
+      if (tooLong) {
+        setError(t('view_W3NCreateForm_long', maxLength));
+        return;
+      }
+
+      const unexpected = name.match(/[^.a-z0-9_-]/);
+      if (unexpected) {
+        setError(t('view_W3NCreateForm_unexpected', unexpected[0]));
+        return;
+      }
+
+      const taken = Boolean(await Web3Names.queryDidForWeb3Name(name));
```

Comment:
Unfortunately this is not always the case. Until we put a fix in place in the blockchain, there could be cases of "dangling" references, in which a DID is deleted by the link between the DID and the web3 name is still on chain. Right now there is nothing we can do about it, but it might be worth to put a FIXME to re-consider this in the next releases, when the mitigation will be in place.

Reply:
Added a TODO!